### PR TITLE
LaunchPad Platform using Ink! Smart Contract

### PR DESCRIPTION
**Pull request by nad128668**
#### Project Abstract

ArtZero & InkWhale team is developing a decentralized LaunchPad using Ink! Smart Contract. All the work will be available for the public to reuse on Github. The project is being built and tested on Aleph Zero network but can be deployed on any parachains or Substrate-based blockchains that support Ink! such as Astar, Aleph Zero and 5ireChain.

After successful Grant Delivery of ArtZero and InkWhale as well as successful launch of NFT Marketplace ArtZero. We would like to create more useful dapps for the ecosystem, not only for us to use but for the public to see how powerfull Ink! Smart Contract is. This decentralized LaunchPad is the first LaunchPad developed using Ink!

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by nad128668** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by nikw3f** [comment]
Hey team, thanks for the application. Could you explain the rationale behind storing Full Project Information on-chain via IPFS rather than other methods? How are you planning to address the potential regulatory challenges, especially concerning token sales and KYC?

Also In the [design](https://www.figma.com/file/63xCCH71Oa8AfJpkK1wCO3/Ink-Whale?type=design&node-id=538-107&mode=design&t=jzPEaYJCUZLjUlDe-0) it says BNB, BUSD tokens. Those are not supported on substrate chains.

File: applications/InkWhale_LaunchPad_using_Ink.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+| 0d.    | Article/Tutorial   | We will write an article or tutorial that explains the work principle as part of the grant. |
+| 1. | Smart Contract Development | We will create 3 main contracts: LaunchPad generator as factory contract to initializer LaunchPad Contract; LaunchPad contract which has all features listed in the Project Details and a general PSP22 token contract for testing purposes|
+| 2. | Backend | We use nodejs and mongodb on AWS Services. The cronjob monitor update queue to make sure data in the database match with data on-chain; The API serves the frontend with following functions: interacting with IPFS network, get launchpad information, get token information, update launchpad information when needed |
+| 3. | Frontend | We will implement the Figma design as in https://www.figma.com/file/63xCCH71Oa8AfJpkK1wCO3/Ink-Whale?type=design&node-id=538-107&mode=design&t=jzPEaYJCUZLjUlDe-0 for the launchpad which contains 11 pages include: Project List; Project Creation (5 sub-pages); Project Details for Project Owner; Project Details for Public; Edit LaunchPad Information; Edit Whitelist and Update Phases;|
+
+
+## Future Plans
```
Comment (nikw3f) [inline applications/InkWhale_LaunchPad_using_Ink.md:122]: Please add some points here

File: applications/InkWhale_LaunchPad_using_Ink.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+
+## Additional Information :heavy_plus_sign:
```
Comment (nikw3f) [inline applications/InkWhale_LaunchPad_using_Ink.md:125]: Please provide total cost. Is there only 1 milestone?

**Review by nikw3f — CHANGES_REQUESTED** [review]

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. I have one quick initial question: Have you looked at https://www.polimec.org/? In general, there are a lot of regulatory problems that you might want to look into. Regarding the project information, I assume you plan to store only hashes on-chain and the data on IPFS, correct? How do you ensure that the data is available in this case? We had a couple of IPFS-related pallet projects in the past, but so far, I don't think any ink! focused projects that tried to overcome potential issues.

File: applications/InkWhale_LaunchPad_using_Ink.md (hunk 2 above)
Comment (nad128668) [reply applications/InkWhale_LaunchPad_using_Ink.md:125]: yes we plan to do everything in 1 milestone

File: applications/InkWhale_LaunchPad_using_Ink.md (hunk 1 above)
Comment (nad128668) [reply applications/InkWhale_LaunchPad_using_Ink.md:122]: you mean points for the Future Plans?

**Comment by nad128668** [comment]
> Thanks for the application. I have one quick initial question: Have you looked at https://www.polimec.org/? In general, there are a lot of regulatory problems that you might want to look into. Regarding the project information, I assume you plan to store only hashes on-chain and the data on IPFS, correct? How do you ensure that the data is available in this case? We had a couple of IPFS-related pallet projects in the past, but so far, I don't think any ink! focused projects that tried to overcome potential issues.

We packed all information to a json file then upload to IPFS server to get the HASH, this Hash is updated to the contract. This method reduces a lot of storage fee when saving data on-chain.

Regarding legal matters, There are 2 KYC processes in the platform and we use BlockPass as third-party KYC service.
- Firstly, KYC is required for project creators. After submitting the project, the project is in disabled mode and our team will send a KYC link to the project owner via email. After KYC is completed, the project will be enabled by our admin.
- Secondly, project owner can choose to KYC all participants or not by enabling Whitelist feature in the platform. It means the project will be in Private Mode, and Public Mode is disabled. Only users in whitelist can participate in the sale. Project owner has to add KYCed addresses via our Whitelist manager. This KYC process is optional for the project owner.

**Comment by nad128668** [comment]
> Hey team, thanks for the application. Could you explain the rationale behind storing Full Project Information on-chain via IPFS rather than other methods? How are you planning to address the potential regulatory challenges, especially concerning token sales and KYC?
> 
> Also In the [design](https://www.figma.com/file/63xCCH71Oa8AfJpkK1wCO3/Ink-Whale?type=design&node-id=538-107&mode=design&t=jzPEaYJCUZLjUlDe-0) it says BNB, BUSD tokens. Those are not supported on substrate chains.

Please check the answer above. 

For the design question, I will ask the designer to update the token symbol, but in our actual implementation, we already change the token symbol to AZERO

**Comment by nikw3f** [comment]
Thanks, i had some more queries:

- Given that this is the first LaunchPad developed using Ink!, how do you plan to educate and onboard potential users and project owners?
- How do you foresee the LaunchPad enhancing the adoption of Ink! Smart Contracts in the broader blockchain ecosystem?
- While you mentioned that an external audit of the code is out of scope for this proposal, when do you anticipate having the code audited, and have you identified potential auditors?

**Comment by nad128668** [comment]
> Thanks, i had some more queries:
> 
> * Given that this is the first LaunchPad developed using Ink!, how do you plan to educate and onboard potential users and project owners?
We are well known in Aleph Zero ecosystem and we already in contact with several projects to launch on our platform. Fyi: our engineers are also working on kyc integration which allow project owners to perform the checking and whitelist the kyced address. I think this feature will be one of our greatest usp as even pinksale doesnt have it yet. Our marketing team has planned a series of contents to educate users about all features of the platform and it will be on our social media such as X (twitter), medium, telegram and discord
> * How do you foresee the LaunchPad enhancing the adoption of Ink! Smart Contracts in the broader blockchain ecosystem?
We have queries from several project owners who are lack of technical skills and dev team and they want to launch projects such as meme coins. These kind of projects will be the ones which needs the launchpad the most at the beginning of the platform launch. However I believe the launchpad will reduce a lot of dev cost for other project owners so they can focus resources on their core product. We plan to deploy the platform on Aleph Zero and then 5ireChain and other Substrate based blockchains in the near future.
> * While you mentioned that an external audit of the code is out of scope for this proposal, when do you anticipate having the code audited, and have you identified potential auditors?
Brushfam did the audit for our nft marketplace contracts and we are already working closely with them to do contract audit for the launchpad

**Comment by nad128668** [comment]
The figma design has been updated with new token symbol, our design team is working on adding kyc feature to the UI

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Thanks for the application @nad128668 while I was willing to support the [initial](https://github.com/w3f/Grants-Program/pull/1342) level 2 PoC for the applications, I'm personally not sure if this time is the best fit for grants program, seeing as you have already had a private sale and distribution of your token. Ideally, you would look for alternative funding sources from the community at this point, such as your own treasury, or you could even try the [Aleph Zero](https://alephzero.org/ecosystem-funding-program) grants program. Please note [our policy](https://github.com/w3f/Grants-Program#guidelines) when it comes to projects who have conducted a token sale.

Also, in your previous grant you [already created](https://github.com/w3f/Grants-Program/pull/1342/files#diff-304355b423aebd1a776bdc4d5fe51352cbd011b43500323a75075b3bbb7d6859R131) a PSP22 contract, as well as a token generator contract, so I would think it wouldn't be needed at this stage?

**Comment by nad128668** [comment]
> our policy

Hi @keeganquigley

Thanks for your feedbacks. We are trying to keep the team going in this tough bear market by getting supported from multiple sources. We think that we have to keep building missing pieces with high quality for the ecosystem to be able to  get the support. The token sale from the public was not enough for the team of 12 people in the longer term and also this LaunchPad project is considered more like a different project from our initial idea of a staking platform.

To be honest, We are already part of EFP and got a grant from Aleph Zero but its for Marketing activities and Legal setup. Our marketing team is working hard to promote the platform as well as attracting more developers in Vietnam to use Ink! Smart Contract. We use the grant to sponsor the Polkadot hackathon in Vietnam for ink! smart contract track and do a lot of promotion and PR campaigns. The reason we applied for W3F grant because we think the technical development of this project will serve a broader range of users, not only from Aleph Zero but also Polkadot and all Substrate-based blockchains. From the beginning, we always keep all of our source code open for the public and I believe many have been using our code for their development. We can separate this project out of Ink Whale platform using different brand but we choose not to do so. ArtZero and Ink Whale already has trust in the ecosystem, our code will be more accessible for the dev.

Regarding the token question, we don't need token generator in the launchpad, we only create LaunchPad generator as factory contract, LaunchPad Core Contract and the token contract is for testing purposes. The token contract also requires a very small work to compare with other 2 contracts.

**Review by semuelle — COMMENTED** [review]
Thanks for the application, @nad128668. I remember evaluating part of your previous grants, and I appreciate your open source work. That being said, I think a platform to launch fungible tokens isn't what we as a technical program should be looking for. I understand that you are trying to lower technical barriers to the ecosystem, but we are not looking to attract low-effort token projects either.

**Comment by takahser** [comment]
@nad128668 due to the issues already raised by other members of the W3F grants committee, namely regulatory concerns, overlaps with existing implementations, concerns related to data availability and your past token sale we concluded that this proposal might not be a good fit for our program. Therefore, I'm going to close this pull request. On behalf of the committee I'd like to thank you for the interest in our program and would like to add that you're welcome to apply for another grant again in the future. Thanks for your understanding, we wish you all the best for your project!

**Pull request closed without merging**